### ISSUE TITLE
Cut area from pdf

### ISSUE THREAD

**Issue by Ninja-Koala** [xournalpp/xournalpp#2392]
It would be nice to be able to cut some area from a pdf onto another page, instead of opening a whole new page.
For example when a pdf contains a diagram, or if you want to extract an exercise from an exercise sheet.

**Comment by rolandlo** [comment]
Can't you just use a screenshot tool for that purpose?

**Comment by Ninja-Koala** [comment]
Yes, but i would find it quite handy to be able to do that from within this tool.
Also: (maybe a bit pedantic) screenshots are raster graphics, but i would like to keep it as vector graphic.
It would be very nice if one could use the "Select Rectangle" tool for this, just like it works on Latex formulas, for example.

**Comment by LittleHuba** [comment]
We are unable to provide vector graphics of partial PDFs as we use a library that does PDF handling. Cutting PDFs is no trivial thing. 

But to help you out: A feature I actually wanted for quite some time is to not use PDFs as background of a page but treat them as element that can be resized and moved. Cropping could also be realized that way.

**Comment by kljohann** [comment]
I prototyped this two years ago using raster graphics (which are sufficient for my use case) and have been using it in my local build of xournalpp since. It's pretty convenient: Basically a variant of the "select rectangle" tool that will insert an image containing a "screenshot" of the selection (which can then be moved around). I would love to have a similar feature upstream (with or without vector grapics) so that I do not have to maintain my own build. I can submit my (simple) implementation as a starting point.
![screenshot_rect](https://user-images.githubusercontent.com/67211/111914405-0f263880-8a72-11eb-92ae-3b79e84c43a8.gif)

**Comment by Alexmaimart** [comment]
Hello! I was looking for this function in Xournal. Can you send your implementation, please?
